### DISCUSSION TITLE
does enabledFreeze prevent only UI from re-rendering ?

### DISCUSSION THREAD

**Discussion by chanphiromsok** [software-mansion/react-native-screens#2094] [Q&A] [answered]
- I wonder why the event emitter is fired in the previous screen
- EvenListener in `SCREEN A` I move to `SCREEN B` -> when even is fire EventListener in `SCREEN A` is fired but `SCREEN A`  is the previous screen
- I have set enabledFreeze and enabledScreen using native-stack 
- So enabledFreeze only prevents from re-render UI? Am I right?

**Comment by tboba** [comment, marked as answer]
Hi @chanphiromsok, yes that's right. When you have react-freeze enabled in your application, JS still can perform operations on the previous screens, but UI on those screens does not update! You can take a look how does it work on TestsExample -> TestFreeze (replace Test42 to TestFreeze in App.tsx to make it work).

Let me know if you have any more questions 🙏
Cheers

**Reply by chanphiromsok** [reply]
> Hi @chanphiromsok, yes that's right. When you have react-freeze enabled in your application, JS still can perform operations on the previous screens, but UI on those screens does not update! You can take a look how does it work on TestsExample -> TestFreeze (replace Test42 to TestFreeze in App.tsx to make it work).
> 
> 
> 
> Let me know if you have any more questions 🙏
> 
> Cheers

Thanks you for confirmation 😍
